core/models/provider.py

```diff
@@ -149,12 +150,73 @@ def is_active(self):
     def get_location(self):
         return self.location
 
+    def get_credential(self, key):
+        cred = self.providercredential_set.filter(key=key)
+        return cred[0].value if cred else None
+
     def get_credentials(self):
         cred_map = {}
         for cred in self.providercredential_set.all():
             cred_map[cred.key] = cred.value
         return cred_map
 
+    def get_routers(self):
+        """
+        Using the two kwargs:
+        'router_name' and 'public_routers'
+        return a list of routers:
+        ['router1']
+        ['router1','router2,'...]
+        """
+        public_routers = self.get_credential('public_routers')
+        router_name = self.get_credential('router_name')
+        if public_routers:
+            return public_routers.split(',')
+        else:
+            return [router_name]
+
+    def select_router(self, router_distribution=None):
+        if not router_distribution:
```

Comment:
Not quite, a router distribution is the _current count of users to router_.

The 'select_router' function is designed to place the user on the router with the *smallest* count to get a 'roughly equal' distribution.

It is able to be passed in as a kwarg to avoid calling `get_router_distribution()` on each iteration of the admin script.